src/content_scripts/contentlib.js

```diff
@@ -31,7 +31,7 @@ export async function waitForElements(
   let error;
   for (let i = 0; i < retries; i++) {
     try {
-      return await Promise.all(queries.map(query => queryElement(query, node)));
+      return queries.map(query => queryElement(query, node));
```

Comment:
I don't get why you no longer await the promises?

Reply:
Because they're not promises (anymore).